### PR TITLE
[RFC-0006] Add CDEvent Receiver Support

### PR DESCRIPTION
**Pull request by adamkenihan**
Adding CDEvent Receiver implementation according to RFC-0006

Fix: #769

### CODE REVIEW

File: internal/server/receiver_handlers.go [hunk 1]
```diff
@@ -197,6 +198,40 @@ func (s *ReceiverServer) validate(ctx context.Context, receiver apiv1.Receiver,
 
 		logger.Info(fmt.Sprintf("handling GitLab event: %s", event))
 		return nil
+	case "cdevents":
+		event := r.Header.Get("Ce-Type")
+		b, err := io.ReadAll(r.Body)
+
+		logger.Info(string(b))
```
[the remark this answers is not in the document]
Comment (adamkenihan) [reply internal/server/receiver_handlers.go]: Yes it was a debug statement, apologies.

File: internal/server/receiver_handlers.go [hunk 2]
```diff
@@ -197,6 +198,40 @@ func (s *ReceiverServer) validate(ctx context.Context, receiver apiv1.Receiver,
...
+		}
+		cdevent, err := cdevents.NewFromJsonBytes(b)
+		if err != nil {
+			return fmt.Errorf("unable to validate event1: %s", err)
+		}
+		err = cdevents.Validate(cdevent)
+		if err != nil {
+			return fmt.Errorf("unable to validate event: %s", err)
+		} else {
+			logger.Info("Validated CDEvent")
```
[the remark this answers is not in the document]
Comment (adamkenihan) [reply internal/server/receiver_handlers.go]: Not needed, removing it is fine.

File: internal/server/receiver_handler_test.go [hunk 3]
```diff
@@ -116,6 +116,52 @@ func Test_handlePayload(t *testing.T) {
...
+			},
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "token",
+				},
+				Data: map[string][]byte{
+					"token": []byte("token"),
+				},
+			},
+			expectedResponseCode: http.StatusOK,
```
[the remark this answers is not in the document]
Comment (adamkenihan) [reply internal/server/receiver_handler_test.go:163]: Added those 2 suggested test cases

File: internal/server/receiver_handlers.go [hunk 4]
```diff
@@ -197,6 +198,40 @@ func (s *ReceiverServer) validate(ctx context.Context, receiver apiv1.Receiver,
 
 		logger.Info(fmt.Sprintf("handling GitLab event: %s", event))
 		return nil
+	case "cdevents":
```
[the remark this answers is not in the document]
Comment (adamkenihan) [reply internal/server/receiver_handlers.go:201]: Yes, will need to add this.

**Comment by stefanprodan** [comment]
@adamkenihan can you please signoff all your commits, this is a requirement for Flux contributions. Also can you run `make test` before pushing changed upstream, by the looks of it, this code does not compile since it imports packages not present in go.mod.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @adamkenihan 🥇

**Pull request merged**